Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetSelectionModel.cpp

```diff
@@ -228,7 +228,7 @@ namespace AzToolsFramework
             return selection;
         }
 
-        AssetSelectionModel AssetSelectionModel::SourceAssetTypeSelection(const QString& pattern, bool multiselect)
+        AssetSelectionModel AssetSelectionModel::SourceAssetTypeSelection(const QRegExp& pattern, bool multiselect)
```

Comment:
If the separate PR (https://github.com/o3de/o3de/pull/8059) for regular expression support instead of string/wildcard gets merged first, then you can pull that in so that it doesn't show up on the diff.